### PR TITLE
Add error handling around parsing for form-encoded data [CH 579]

### PR DESCRIPTION
**Pull request by chasingSublimity**
Pretty simple change, more info on the [ticket](https://app.clubhouse.io/active-prospect/story/579/improve-error-handling-for-malformed-form-encoded-data-in-the-standard-inbound-integration)

### CODE REVIEW

File: src/inbound.coffee [hunk 1]
```diff
@@ -81,7 +81,11 @@ request = (req) ->
 
       # if form URL encoding, convert dot notation keys
       if mimeType == 'application/x-www-form-urlencoded'
-        parsed = flat.unflatten(parsed)
+        try
+          parsed = flat.unflatten(parsed)
+        catch e
+          formEncodedError = e.toString()
+          throw new HttpError(400, {'Content-Type': 'text/plain'}, "Body does not contain form-encoded data or form-encoded data is unparseable -- #{formEncodedError}.")
```
Comment (cgrayson) [inline src/inbound.coffee]: Minor but I'd shorten and simplify that message, maybe just something like "Unable to parse body" (plus keep the error string, which gives a better hint to the specific problem).

Comment (chasingSublimity) [reply src/inbound.coffee]: You don't think it's helpful to include the mimetype in the message? I'm down to shorten, but it seems like indicating that it the integration thinks that the original form data might be nice. Either way is fine by me, just curious. 👍

Comment (cgrayson) [reply src/inbound.coffee]: Your call. I was thinking it was a little bit of a red herring, actually, in that (afaik) you can't form-encode this example input to fix the problem. But maybe if a source is mistakenly using the _wrong_ mime-type, your message would tip them off to that? (Not even sure now what the fix would be here, actually… 🤔 ).

Comment (chasingSublimity) [reply src/inbound.coffee]: Yea, i guess in this test case the encoding is valid, it's the json parsing that bonks. That makes sense to me. 👍

**Pull request merged**